rsfbclient-rust/src/client.rs

```diff
@@ -455,7 +459,18 @@ impl FirebirdWireConnection {
         )?)?;
         self.socket.flush()?;
 
-        let (op_code, mut resp) = self.read_packet()?;
+        let (mut op_code, mut resp) = self.read_packet()?;
+
+        // Read lazy responses
+        for _ in 0..self.lazy_count {
+            if op_code != WireOp::Response as u32 {
+                return err_conn_rejected(op_code);
+            }
+            self.lazy_count -= 1;
+            parse_response(&mut resp)?;
+
+            op_code = resp.get_u32()?;
+        }
```

Comment:
@jairinhohw I think you probably authored this-

Could you elaborate a bit on the lazy response reading?

some specific questions:
1. Is the goal to help avoid syscalls?
2. Maybe a naive question, which applies not only here- does the ordering of response XDR structures by fb server necessarily match TCP ordering of API calls? Do docs say somewhere that this is so? 

any additional info is also appreciated